### ISSUE TITLE
Request should return Promise.reject when response code is not 2xx

### ISSUE THREAD

**Issue by georgehrke** [owncloud/davclient.js#24]
Right now the developer has to check if the request was successful in the then "block".
It would be easier for developers, if instead of resolving, davclient.js would automatically reject the promise if the response code is not a 2xx one.

**Comment by PVince81** [comment]
Indeed. I was also confused by this for a while until I remembered.
